backend/models/user.js

```diff
@@ -0,0 +1,28 @@
+const mongoose = require('mongoose');
+
+const userSchema = new mongoose.Schema({
+  firstName: { type: String, required: true },
+  lastName: { type: String, required: true },
+  email: { type: String, required: true, unique: true },
+  password: { type: String, required: true },
+  role: { type: String, required: true },
+  gender: { type: String },
+  phone: { type: String },
+  country: { type: String },
+  state: { type: String },
+  city: { type: String },
+  zip: { type: String },
+  birthDate: { type: String },
+  createdDate: { type: Date, default: Date.now },
```

Comment:
Add 'timestamps' as parameter in Schema and mongoose will add createdAt and updatedAt fields automatically.
```
new Schema(
{....},
  {
    timestamps: true,
  },
)
```